File: generated_code/python/solution_3379_1.py
class Solution:
    def scoreOfString(self, s: str) -> int:
        """
        Calculates the score of a string, defined as the sum of the absolute
        differences between the ASCII values of adjacent characters.
        
        Time Complexity: O(N) where N is the length of the string s.
        Space Complexity: O(1) as we only use a few integer variables.
        """
        score = 0
        # Iterate through the string, comparing each character with the next one
        for i in range(len(s) - 1):
            score += abs(ord(s[i]) - ord(s[i + 1]))
            
        return score

File: generated_code/cpp/solution_3379_1.cpp
#include <string>
#include <cmath>
#include <numeric>

using namespace std;

class Solution {
public:
    /**
     * Algorithm: Linear Scan / One-pass
     * Time Complexity: O(n), where n is the length of string s. 
     *                  We traverse the string exactly once.
     * Space Complexity: O(1), as we only use a single integer to accumulate the score.
     */
    int scoreOfString(string s) {
        int score = 0;
        
        // Iterate from the first character to the second-to-last character
        // to compare adjacent pairs (s[i] and s[i+1]).
        for (size_t i = 0; i < s.length() - 1; ++i) {
            // Add the absolute difference of ASCII values of adjacent characters.
            // abs() handles the magnitude regardless of character order.
            score += abs(s[i] - s[i + 1]);
        }
        
        return score;
    }
};